sleap/skeleton.py

```diff
@@ -999,34 +999,153 @@ def to_json(self, node_to_idx: Optional[Dict[Node, int]] = None) -> str:
+            edges_dicts.append(
+                {
+                    # Note: Insert idx is not the same as the edge index in the skeleton
+                    # in legacy SLEAP.
+                    "edge_insert_idx": edge_ind,
+                    "key": 0,  # Always 0.
+                    "source": {"py/object": "sleap.skeleton.Node", "py/state": {"name": source.name, "weight": 1.0}},
+                    "target": {"py/object": "sleap.skeleton.Node", "py/state": {"name": target.name, "weight": 1.0}},
+                    # "target": {"py/id": node_to_id[target]},
+                    "type": edge_type,
+                }
+            )
+        print(f'edges_dicts: {edges_dicts}')
+
+        # Build links dicts for symmetry edges.
+        for symmetry_ind, symmetry in enumerate(self.symmetries):
+            print(f'symmetry_ind: {symmetry_ind}')
+            print(f'symmetry: {symmetry}')
+            if symmetry_ind == 0:
+                edge_type = {
+                    "py/reduce": [
+                        {"py/type": "sleap.skeleton.EdgeType"},
+                        {"py/tuple": [2]},  # 1 = real edge, 2 = symmetry edge
+                    ]
+                }
+            else:
+                edge_type = {"py/id": 2}
+
+            src, dst = tuple(symmetry.nodes)
+            print(f'src: {src}')
+            print(f'dst: {dst}')
+            edges_dicts.append(
+                {
+                    "key": 0,
+                    "source": {"py/id": node_to_id[src]},
+                    "target": {"py/id": node_to_id[dst]},
+                    "type": edge_type,
+                }
+            )
+        # Create graph field
+        graph = {
+            "name": self.name,
+            "num_edges_inserted": len(self.edges),
+        }
+        # Create skeleton dict
         if self.is_template:
-            data = {
-                "nx_graph": graph,
+            # Template skeletons have additional fields
+            nx_graph = {
+                "directed": True,
+                "graph": graph, 
+                "links": edges_dicts,
+                "multigraph": True,
+                # In the order in Skeleton.nodes and must match up with nodes_dicts.
+                "nodes": [{"id": {"py/id": node_to_id[node]}} for node in self.nodes],
+            }
+            skeleton_dict = {
                 "description": self.description,
+                "nx_graph": nx_graph, 
                 "preview_image": self.preview_image,
             }
         else:
-            data = graph
-
-        json_str = jsonpickle.encode(data)
-
+            skeleton_dict ={
+                    "directed": True,
+                    "graph": graph,
+                    "links": edges_dicts,
+                    "multigraph": True,
+                    # In the order in Skeleton.nodes and must match up with nodes_dicts.
+                    "nodes": [{"id": {"py/id": node_to_id[node]}} for node in self.nodes],}
```

Comment:
Assuming jsonpickle is decoding by reading things in order, we can assume that all `Node`s will only appear in the `"nodes"` list once. Therefore, we can (hopefully) move the `"nodes"` list before the `"links"` dictionary. The `"nodes"` list would need to define the nodes in the `{py/object: ..., py/state: ...}` dictionary. Then, the `"links"` dictionary should just reference nodes by `"py/id"` (referencing the `Node`'s index in the `"nodes"` list).
```suggestion
                    "nodes": [{"id": {"py/object": "sleap.skeleton.Node", "py/state": {"name": node.name, "weight": node.weight}}} for node in self.nodes],
                    "links": edges_dicts,
                    "multigraph": True,
```